pkg/networkdevice/profile/profiledefinition/metrics.go

```diff
@@ -0,0 +1,150 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package profiledefinition
+
+import (
+	"regexp"
+)
+
+type ProfileMetricType string
+
+const (
+	ProfileMetricTypeGauge                 ProfileMetricType = "gauge"
+	ProfileMetricTypeMonotonicCount        ProfileMetricType = "monotonic_count"
+	ProfileMetricTypeMonotonicCountAndRate ProfileMetricType = "monotonic_count_and_rate"
+	ProfileMetricTypeRate                  ProfileMetricType = "rate"
+	ProfileMetricTypeFlagStream            ProfileMetricType = "flag_stream"
+
+	// ProfileMetricTypeCounter is DEPRECATED
+	// `counter` is deprecated in favour of `rate`
+	ProfileMetricTypeCounter ProfileMetricType = "counter"
+
+	// ProfileMetricTypePercent is DEPRECATED
+	// `percent` is deprecated in favour of `scale_factor`
+	ProfileMetricTypePercent ProfileMetricType = "percent"
+)
+
+// SymbolConfig holds info for a single symbol/oid
+type SymbolConfig struct {
+	OID  string `yaml:"OID" json:"OID"`
+	Name string `yaml:"name" json:"name"`
+
+	ExtractValue         string         `yaml:"extract_value" json:"extract_value"`
+	ExtractValueCompiled *regexp.Regexp // TODO: ADD `-` yaml/json annotation?
+
+	MatchPattern         string         `yaml:"match_pattern" json:"match_pattern"`
+	MatchValue           string         `yaml:"match_value" json:"match_value"`
+	MatchPatternCompiled *regexp.Regexp // TODO: ADD `-` yaml/json annotation?
+
+	ScaleFactor      float64 `yaml:"scale_factor" json:"scale_factor"`
+	Format           string  `yaml:"format" json:"format"`
+	ConstantValueOne bool    `yaml:"constant_value_one" json:"constant_value_one"`
+
+	// `metric_type` is used for force the metric type
+	//   When empty, by default, the metric type is derived from SNMP OID value type.
+	//   Valid `metric_type` types: `gauge`, `rate`, `monotonic_count`, `monotonic_count_and_rate`
+	//   Deprecated types: `counter` (use `rate` instead), percent (use `scale_factor` instead)
```

Comment:
Confusing that comment says to use `scale_factor` but `scale_factor` is not listed in the line above as a valid `metric_type`.

Reply:
I think Dan summarised it well :)